### PR TITLE
Gen match dr

### PR DESCRIPTION
**Pull request by jalimena**
add the dR between a particle and its gen match
then add these dR histograms for electrons and muons

### CODE REVIEW

**Comment by jalimena** [comment]
Tested in our mumu channel on a few events. Given the histogram below (and also printouts from InfoPrinter), this PR works as expected - notice the stats box.

Screen Shot 2020-02-10 at 10 12 21 AM

**Comment by jalimena** [comment]
i believe this PR can be merged

**Comment by btcardwell** [comment]
Looks good to me, though it's probably best for Brian to take a quick look as well.

File: Configuration/python/histogramDefinitions.py [hunk 1]
```diff
@@ -522,6 +528,12 @@
             binsX = cms.untracked.vdouble(getPdgBins(["unmatched", "quarks", "leptons", "bosons"])),
             inputVariables = cms.vstring("abs (genMatchedParticle.bestMatchPdgId)"),
             ),
+        cms.PSet (
+            name = cms.string("electronBestMatchDeltaR"),
+            title = cms.string(";#DeltaR between electron and generator particle matched to electron"),
+            binsX = cms.untracked.vdouble(300,0,6),
+            inputVariables = cms.vstring("genMatchedParticle.dR"),
+        )
```
Comment (btcardwell) [inline Configuration/python/histogramDefinitions.py]: Missed a comma here

Comment (jalimena) [reply Configuration/python/histogramDefinitions.py]: thanks, fixed now!

**Comment by jalimena** [comment]
ping @bpf6qc

**Comment by bpf6qc** [comment]
Sorry for the delay in responding to this. I have some comments.

1. `dR` should be initialized to INVALID_VALUE for instead of zero, since zero could indicate a valid match.
2. `dR` should be named instead `bestMatchDR` or similar to indicate that it's the dR only to the `bestMatch`.
3. A bigger question -- do we need this? `genMatchedParticle.bestMatchPdgId` seems like a convenience already given that you could do `genMatchedParticle.bestMatch.pdgId` and equivalently:

```deltaR(muon.genMatchedParticle.bestMatch, muon)```

Although as I type that out, I see this could be confusing with an inputCollection of `("muons, "muons")` and looping over things twice.

In short for this third one if the comment is "we'd like this convenience", I'm happy to merge this.

**Comment by jalimena** [comment]
thanks, @bpf6qc . 

1 and 2 should now be done.

for 3, I believe we tried what you suggest, but it didn't work. I didn't see another way to do this (and believe me, I tried)
